test/MigrationsConfigurationFactoryTest.php

```diff
@@ -0,0 +1,86 @@
+use PHPUnit\Framework\TestCase;
+use Psr\Container\ContainerInterface;
+use Roave\PsrContainerDoctrine\AbstractFactory;
+use Roave\PsrContainerDoctrine\MigrationsConfigurationFactory;
+
+final class MigrationsConfigurationFactoryTest extends TestCase
+{
+    public function testExtendsAbstractFactory() : void
+    {
+        $this->assertInstanceOf(AbstractFactory::class, new MigrationsConfigurationFactory());
+    }
+
+    public function testConfigValues() : void
+    {
+        $connection = $this->buildConnection();
+        $container  = $this->createMock(ContainerInterface::class);
+
+        $config = [
+            'doctrine' => [
+                'migrations_configuration' => [
+                    'orm_default' => [
+                        'directory' => 'test/TestAsset',
+                        'name'      => 'Foo Bar',
+                        'namespace' => 'Acme\Lib\Migrations',
+                        'table'     => 'baz',
+                        'column'    => 'ver',
+                    ],
+                ],
+            ],
+        ];
+
+        $container->expects($this->exactly(2))
+            ->method('has')
+            ->withConsecutive(['config'], ['doctrine.connection.orm_default'])
+            ->willReturnOnConsecutiveCalls(true, true);
+
+        $container->expects($this->exactly(2))
+            ->method('get')
+            ->withConsecutive(['config'], ['doctrine.connection.orm_default'])
+            ->willReturnOnConsecutiveCalls($config, $connection);
+
+        $migrationsConfiguration = (new MigrationsConfigurationFactory())($container);
+
+        $this->assertSame($connection, $migrationsConfiguration->getConnection());
+        $this->assertSame(
+            $config['doctrine']['migrations_configuration']['orm_default']['directory'],
+            $migrationsConfiguration->getMigrationsDirectory()
+        );
+        $this->assertSame(
+            $config['doctrine']['migrations_configuration']['orm_default']['name'],
+            $migrationsConfiguration->getName()
+        );
+        $this->assertSame(
+            $config['doctrine']['migrations_configuration']['orm_default']['namespace'],
+            $migrationsConfiguration->getMigrationsNamespace()
+        );
+        $this->assertSame(
+            $config['doctrine']['migrations_configuration']['orm_default']['table'],
+            $migrationsConfiguration->getMigrationsTableName()
+        );
+        $this->assertSame(
+            $config['doctrine']['migrations_configuration']['orm_default']['column'],
+            $migrationsConfiguration->getMigrationsColumnName()
+        );
+    }
+
+    private function buildConnection() : Connection
+    {
+        $schemaManager = $this->createMock(AbstractSchemaManager::class);
+        $platform      = $this->createMock(AbstractPlatform::class);
+        $connection    = $this->createPartialMock(Connection::class, ['getSchemaManager', 'getDatabasePlatform']);
```

Comment:
Why a partial mock?

Reply:
To follow the same approach of `EntityManagerFactoryTest`: only leverage used methods

Reply:
A partial mock does not prevent usage of other methods: I'd say that the other tests are likely broken too. They were ported from the previous version, so there is indeed legacy to be fixed in some future.

Reply:
So we go for simple mocks?

Reply:
Yup :+1: